test/general/test_entrance_rando.py

```diff
@@ -99,6 +103,28 @@ def test_ordered_targets(self):
         group_order = [prev := group.randomization_group for group in retrieved_targets if prev != group.randomization_group]
         self.assertEqual([ERTestGroups.TOP, ERTestGroups.BOTTOM], group_order)
 
+    def test_selective_dead_ends(self):
+        """test that entrances that EntranceLookup has not been told to consider are ignored when finding dead-ends"""
+        multiworld = generate_test_multiworld()
+        generate_disconnected_region_grid(multiworld, 5)
+        exits_set = set([ex for region in multiworld.get_regions(1)
+                        for ex in region.exits if not ex.connected_region
+                         and ex.name != "region20_right" and ex.name != "region21_left"])
+
+        lookup = EntranceLookup(multiworld.worlds[1].random, coupled=True, usable_exits=exits_set)
+        er_targets = [entrance for region in multiworld.get_regions(1)
+                      for entrance in region.entrances if not entrance.parent_region and
+                      entrance.name != "region20_right" and entrance.name != "region21_left"]
+        for entrance in er_targets:
+            lookup.add(entrance)
+
+        dead_end_region = multiworld.get_region("region20", 1)
+        for dead_end in dead_end_region.entrances:
+            if dead_end.name == "region20_top":
+                break
+        # there should be only this one dead-end
```

Comment:
It wasn't immediately clear to me why this should be the case (because this region has only a top and right exit and the right has been pulled out of the set) so perhaps additional exposition may be in order